### ISSUE TITLE
Web: DateInput onInput: Event is not InputEvent

### ISSUE THREAD

**Issue by hfhbd** [JetBrains/compose-jb#635]
If you have:
```kotlin
Input(type = InputType.DateTimeLocal, attrs = {
    onInput {
        val event = it.nativeEvent
    }
})
```
then `onInput` will always fail with a `ClassCastException`, because https://github.com/JetBrains/androidx/blob/107c4ac00bb4c99f29b203d71738ae7cd00c6c75/compose/web/src/jsMain/kotlin/androidx/compose/web/attributes/WrappedEventListener.kt#L150
will always cast it to `InputEvent`.

In Chrome and Safari, the `event` is type of `Event`, not `InputEvent`.
<img width="1920" alt="Bildschirmfoto 2021-05-04 um 19 24 00" src="https://user-images.githubusercontent.com/22521688/117044249-86f0af80-ad0e-11eb-8dd2-9f5df8897228.png">
<img width="1920" alt="Bildschirmfoto 2021-05-04 um 19 23 47" src="https://user-images.githubusercontent.com/22521688/117044263-89eba000-ad0e-11eb-880e-827fdcfd21cf.png">

Safari: Version 14.1 (16611.1.21.161.6)
Chrome: Version 90.0.4430.93 (Offizieller Build) (x86_64)
macOS: 11.3.1 (20E241)

Sample Project: https://github.com/hfhbd/ComposeTodo

**Comment by eymar** [comment]
Hi @hfhbd, thanks for reporting this issue.

it's possible to workaround this problem:

```koltin
Input(
        type = InputType.DateTimeLocal,
        attrs = {
            addEventListener("input") { 
                val event = it.nativeEvent // nativeEvent is of type `Event `
            }
        }
    )
```
Let me know if this helps.

In general, we want to have accurately typed events. So we'll keep on improving it.

**Comment by hfhbd** [comment]
@eymar Thanks, it helps, was to easy to find :)

**Comment by eymar** [comment]
fixed in 0.5.0-build228

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6403">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
